doc/examples/reconst_fwdti.py

```diff
@@ -0,0 +1,185 @@
+model is not specific to particular type of tissue. For example, diffusion
+anisotropy in regions near the cerebral ventricle and parenchyma can be
+underestimated by partial volume effects of the cerebral spinal fluid (CSF).
+This free water contamination can particularly corrupt diffusion tensor imaging
+analysis of microstructural changes when different groups of subject show
+different brain morphology (e.g. brain ventricle enlargement associated with
+brain tissue atrophy that occurs in several brain pathologies and ageing).
+
+A way to remove this free water influences is to expand the DTI model to take
+into account an extra compartment representing the contributions of free water
+diffusion. The expression of the expanded DTI model is shown below:
+
+.. math::
+
+    S(\mathbf{g}, b) = S_0(1-f)e^{-b\mathbf{g}^T \mathbf{D}
+    \mathbf{g}}+S_0fe^{-b D_{iso}}
+
+where $\mathbf{g}$ and $b$ are diffusion grandient direction and weighted
+(more information see :ref:`example_reconst_dti`), $S(\mathbf{g}, b)$ is the
+diffusion-weighted signal measured, $S_0$ is the signal conducted in a
+measurement with no diffusion weighting, $\mathbf{D}$ is the diffusion tensor,
+$f$ the volume fraction of the free water component, and $D_iso$ is the
+isotropic value of the free water diffusion (normally set to $3.0 \times
+10^{-3} mm^{2}s^{-1}$).
+
+In this example, we show how to process a diffusion weighted dataset using the
+free water elimantion.
+
+Let's start by importing the relevant modules:
+"""
+
+import numpy as np
+import dipy.reconst.fwdti as fwdti
+import dipy.reconst.dti as dti
+import matplotlib.pyplot as plt
+from dipy.data import fetch_cenir_multib
+from dipy.data import read_cenir_multib
+from dipy.segment.mask import median_otsu
+
+"""
+Without spatial constrains the free water elimination model cannot be solved
+in data acquired from one non-zero b-value _[Hoy2014]. Therefore, here we
+download a dataset that was required from multi b-values.
+"""
+
+fetch_cenir_multib(with_raw=False)
+
+"""
+From the downloaded data, we read only the data acquired with b-values up to
+2000 $s.mm^{-2} to decrease the influence of non-Gaussain _[Hoy2014].
+"""
+
+bvals = [200, 400, 1000, 2000]
+
+img, gtab = read_cenir_multib(bvals)
+
+data = img.get_data()
+
+affine = img.get_affine()
+
+"""
+The free water DTI model can take some minutes to process the full data set.
+Thus, we remove the background of the image to avoid unnecessary calculations.
+"""
+
+maskdata, mask = median_otsu(data, 4, 2, False, vol_idx=[0, 1], dilate=1)
+
+"""
+Moreover, for illustration purposes we process only an axial slice of the
+data.
+"""
+
+axial_slice = 40
+
+mask_roi = np.zeros(data.shape[:-1], dtype=bool)
+mask_roi[:, :, axial_slice] = mask[:, :, axial_slice]
+
+"""
+The free water elimantion model fit can then be initialized by instantiating
+a FreeWaterTensorModel class object:
+"""
+
+fwdtimodel = fwdti.FreeWaterTensorModel(gtab, 'NLS', cholesky=False)
```

Comment:
I just gave a quick run to the example, and since you pass all the keywords, changing it to wls unfortunately crashed the function as cholesky is not an option for this one.

Creating the object works, but fitting gives 
~~~python
In [19]: fwdtifit = fwdtimodel.fit(data, mask=mask_roi)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-3dbd1ad0d42c> in <module>()
----> 1 fwdtifit = fwdtimodel.fit(data, mask=mask_roi)

/home/samuel/python/dipy/reconst/fwdti.py in fit(self, data, mask)
    169         params_in_mask, S0_im = self.fit_method(self.design_matrix,
    170                                                 data_in_mask,
--> 171                                                 *self.args, **self.kwargs)
    172 
    173         if mask is None:

TypeError: wls_fit_tensor() got an unexpected keyword argument 'cholesky'
~~~